libs/wingc/src/jsify.rs

```diff
@@ -352,13 +309,38 @@ impl JSifier {
 						// In either case, it's probably safe to assume that case conversion is required.
 						true
 					};
+
+					let arg_string = self.jsify_arg_list(&args, None, None, needs_case_conversion, phase);
+					let expr_string = match &function.kind {
+						ExprKind::Reference(reference) => self.jsify_reference(reference, Some(needs_case_conversion), phase),
+						_ => format!("({})", self.jsify_expression(function, phase)),
+					};
+
+					if let Some(function_sig) = function_type.as_function_sig() {
+						if let Some(js_override) = &function_sig.js_override {
+							let self_string = &match &function.kind {
+								//
+								ExprKind::Reference(Reference::Identifier(_)) => "global".to_string(),
+								ExprKind::Reference(Reference::NestedIdentifier { object, .. }) => {
+									self.jsify_expression(object, phase).clone()
+								}
+
+								_ => expr_string,
+							};
+							let patterns = &[MACRO_REPLACE_SELF, MACRO_REPLACE_ARGS];
+							let replace_with = &[self_string, &arg_string];
+							let ac = AhoCorasick::new(patterns);
+							return ac.replace_all(js_override, replace_with);
+						}
+					}
+
+					format!("({}{}({}))", auto_await, expr_string, arg_string)
+				} else {
+					panic!(
+						"Unable to call expressions, missing type information: {}",
+						function.span
```

Comment:
```suggestion
					panic!(
						"Unable to call expressions at {}, it's not a closure",
						function.span
```

Reply:
Technically this block would only happen if the `evaluated_type` for the expressions is empty. That being said, this panic does belong a couple of lines above where we do as_function_sig so I added it there. I also slightly improved this panic to look a bit more like your suggestion.